File: python/first_missing_positive.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        numbers = {n for n in nums if n > 0}
        if not numbers:
            return 1
        max_ = max(numbers)
        result = 1
        for i in range(1, max_+2):
            if i not in numbers:
                return i
        return result + 1

File: rust/src/first_missing_positive.rs
use std::collections::HashSet;

impl Solution {
    pub fn first_missing_positive(nums: Vec<i32>) -> i32 {
        let mut numbers: HashSet<i32> = HashSet::with_capacity(nums.len()/2);
        for n in nums {
            if n > 0 {
                numbers.insert(n);
            }
        }
        for i in (1..i32::MAX) {
            if !numbers.contains(&i) {
                return i;
            }
        }
        return i32::MAX;
    }
}
